sklearn/preprocessing/data.py

```diff
@@ -148,15 +154,21 @@ def scale(X, axis=0, with_mean=True, with_std=True, copy=True):
             raise ValueError("Can only scale sparse matrix on axis=0, "
                              " got axis=%d" % axis)
         if with_std:
```

Comment:
Is this check still needed if we have `if with_std not in (...)`?

Reply:
@thomasjpfan Yes, so we can restrict the user to (1, 2, True) values.

Reply:
Yes, it's an additional check.

Reply:
We can raise an error earlier by including the following check right after `X = check_array(...)`

```py
if with_std not in (1, 2, True, False):
    raise ValueError("Invalid value for `with_std`: {}".format(
                     str(with_std)))
```

And remove the two checks we have now.